authorizers/iconik/headers-authorizer.ts

```diff
@@ -1,39 +1,32 @@
 import { getEnv } from '@helper/environment';
-import { errorHandler } from '@helper/error-handler';
 import { log } from '@helper/logger';
 import { IconikParams } from '@workflowwin/iconik-api';
-import { APIGatewayAuthorizerEvent, APIGatewayAuthorizerResult, Handler } from 'aws-lambda';
+import { APIGatewayAuthorizerEvent, Handler } from 'aws-lambda';
 import { getCallerAndOwner } from './helper';
 import { generatePolicy } from '../policy-generator';
-import { IconikContext, IconikEnhancedAuthContext } from './interfaces/context';
+import { AuthorizerResponse, IconikContext, IconikEnhancedAuthContext } from './interfaces/context';
 
-export const iconikAuthorizer: Handler<
-  APIGatewayAuthorizerEvent,
-  (APIGatewayAuthorizerResult & { context: IconikEnhancedAuthContext }) | undefined
-> = async (event, context) => {
+export const iconikAuthorizer: Handler<APIGatewayAuthorizerEvent, AuthorizerResponse> = async (event, context) => {
   log('[Iconik Authorizer]', event);
   try {
     const callerId: string = event['headers']['User-Id'];
     const authToken: string = event['headers']['auth-token'];
 
-    const iconikParams: IconikParams = {
-      authToken,
-      iconikUrl: getEnv('ICONIK_URL'),
-      appId: getEnv('ICONIK_APP_ID'),
-      systemDomainId: getEnv('ICONIK_DOMAIN_ID'),
-    };
+    const iconikUrl = getEnv('ICONIK_URL');
+    const appId = getEnv('ICONIK_APP_ID');
+    const systemDomainId = getEnv('ICONIK_DOMAIN_ID');
+
+    const iconikParams: IconikParams = { authToken, iconikUrl, appId, systemDomainId };
 
-    const { caller, appOwner }: IconikContext = await getCallerAndOwner(iconikParams, callerId);
+    const { appOwner }: IconikContext = await getCallerAndOwner(iconikParams, callerId);
     const context: IconikEnhancedAuthContext = {
       ...iconikParams,
-      callerId: caller.id,
-      callerEmail: caller.email,
-      ownerId: appOwner.id,
-      ownerEmail: appOwner.email,
+      ownerId: appOwner?.id,
+      ownerEmail: appOwner?.email,
     };
 
-    return generatePolicy<IconikEnhancedAuthContext | any>(`user|${caller?.id}`, 'Allow', event.methodArn, context);
+    return generatePolicy<IconikEnhancedAuthContext | any>(`user|${appOwner?.id}`, 'Allow', event.methodArn, context);
   } catch (error) {
-    errorHandler(error);
+    log('[Iconik Authorizer] error: ', error);

```

Comment:
The authorizer should return an error. If iconik returns 401, and you just log it, a lambda function could start working, but it shouldn't because the user is unauthorized. Only if you return the error, API Gateway understands that it shouldn't trigger the lambda function. You can just log it and throw it again.

Reply:
Sorry, I deleted errorHandler when I deleted them from CAs and WHs.